### ISSUE TITLE
Question: handling incomplete graph

### ISSUE THREAD

**Issue by sidney-tio** [EMI-Group/tensorneat#10]
Thanks for the fantastic work on making NEAT in Jax! I'm new to this field and I'm trying to understand the repository better.

How does the repository handle forward computations when the graph is incomplete? For example, in the XOR example, the networks are initialized with 3 inputs and 1 output nodes, but no hidden nodes. In that case, it is quite likely one of the mutations will lead to no inputs to the output (e.g. consecutive del_conns).

It is not clear to me how the code will handle this edge case, so any pointers will be helpful!

**Comment by WLS2002** [comment]
Thank you for your interest in TensorNEAT! I'll explain your question with an example.

![githubissue](https://github.com/user-attachments/assets/7912b430-c23f-4f66-927a-a5d2657122b2)

As shown in the diagram, when the network is initialized (with all three connections present), the output is:

$$
O_1 = \text{Act}(r * \text{Agg}(w_1 * I_1, w_2 * I_2, w_3 * I_3) + b) = \text{Identity}(1 * \text{Sum}(1, 2, 3) + 0) = 6
$$

When the connection with weight = 3 is deleted, the output becomes:

$$
O_1 = \text{Act}(r * \text{Agg}(w_1 * I_1, w_2 * I_2) + b) = \text{Identity}(1 * \text{Sum}(1, 2) + 0) = 3
$$

In an extreme case, if all connections are deleted, the value of $O_1$ will be $\texttt{NaN}$.

In summary, removing edges will not affect the network's overall behavior unless all connections to a node are deleted. If a node has no incoming connections with non- $\texttt{NaN}$ values, its value will also become $\texttt{NaN}$.

If you have further questions, feel free to ask! I'd be happy to help.

**Comment by sidney-tio** [comment]
Thanks for the detailed reply (on thanksgiving weekend too! if you celebrate) For 2D XOR, the likelihood of getting all connections are deleted is significant, so it requires the fitness function to interpret the nan values.

**Comment by WLS2002** [comment]
You’re welcome! 

You’re right, and TensorNEAT already includes handling for `NaN` values in the fitness function. See [this line in the code](https://github.com/EMI-Group/tensorneat/blob/main/src/tensorneat/pipeline.py#L91). So you don’t need to worry about this scenario.
